### PR TITLE
Adds detected language to extracted style tags in TSX

### PR DESCRIPTION
**Pull request by Princesseuh**
## Changes

A long time ago, our language tooling would provide helpful intellisense inside SCSS and LESS tags, this was lost without notice. A recent regression made me remember this, so this PR brings it back

Additionally, this PR changes the script tag extraction a bit to handle `is:raw` and other edge cases better.

## Testing

Added tests

## Docs

N/A

### CODE REVIEW

File: internal/printer/print-to-tsx.go [hunk 1]
```diff
@@ -137,21 +137,57 @@ var htmlEvents = map[string]bool{
...
+	}
+
+	for _, attr := range attrs {
+		if attr.Key == "lang" {
+			if attr.Type == astro.QuotedAttribute {
+				return strings.TrimSpace(strings.ToLower(attr.Val))
+			} else {
+				// If the lang attribute exists, but is not quoted, we can't tell what's inside of it
+				// So we'll just return "unknown" and let the downstream client decide what to do with it
+				return "unknown"
```
Comment (bluwy) [inline internal/printer/print-to-tsx.go:152]: Does this mean if the user writes `<style lang=scss>`, we return `"unknown"`? Shouldn't it be possible to parse the value still per HTML behaviour?

File: packages/compiler/test/tsx/meta.ts [hunk 2]
```diff
@@ -52,6 +52,7 @@ test('extract scripts', async () => {
 				},
 				type: 'module',
 				content: 'console.log({ test: `literal` })',
+				lang: '',
```
[the remark this answers is not in the document]
Comment (Princesseuh) [reply packages/compiler/test/tsx/meta.ts:55]: Go make this super annoying, it serializes nil strings into an empty string instead of undefined/omitting the property (and there's no omitempty, unlike JSON). 

We'd need two separate structs, but then since there's no inheritance, we'd need to duplicate the definition.

In truth, I kinda wanted to use the same format for both so that if there's ever a future where the language of a script tag is decoupled from its hoisting we already have the shape for it and it wouldn't be breaking (ex: `is:inline` + `lang="ts"` would result in type inline, lang ts)

File: internal/printer/print-to-tsx.go (hunk 1 above)
Comment (Princesseuh) [reply internal/printer/print-to-tsx.go:152]: Ah yeah, you're right. I always forget about unquoted values, will fix!

Comment (Princesseuh) [reply internal/printer/print-to-tsx.go:152]: It turns out that for our compiler, unquoted values are included in the `QuotedAttribute` type, so no code change needed here!

File: packages/compiler/test/tsx/meta.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (bluwy) [reply packages/compiler/test/tsx/meta.ts:55]: Thanks that's good to know. Not a big deal then as long as our public API hides it. Maybe we could further generalize it with a `props: { ... }` property too (if Go allows that).

**Review by bluwy — APPROVED** [review]
